python/lsst/ts/dsm/utils.py

```diff
@@ -22,60 +22,83 @@ def convert_time(in_time):
     float
         The TAI time corresponding to the input.
     """
-    ptime = Time(in_time, scale='utc')
+    ptime = Time(in_time, scale="utc")
```

Comment:
In the doc string please document what this converts in_time **to**. What does it return?

Interesting note below. How does that work? Once the line above has run I would expect ptime to be in UTC and that's the end of it -- especially if in_time can only be a float and not an astropy.time.Time.

Reply:
The return value is documented as a float. The in_time parameter is documented as a string. The issue comes trying to derive a float from the astropy.time object. There is no way to get a timestamp (float) without going through the datetime intermediate object. This has the issue that the assumed time zone in the datetime object is the local time zone where the conversion happens. This means it is inconsistent in the timestamp return without doing the replace action as the timestamp will change depending on where the machine doing the conversion is located.

Reply:
Thanks. I missed the doc string for in_time because github didn't show it (nor the brief description) and I forgot to expand the text.

Please expand on what is returned. I think it is TAI unix seconds? If so, consider using salobj's `tai_from_utc` function:
```
return salobj.tai_from_utc(ptime)
```
For what it's worth, astropy 4.2 will have built-in support for `unix_tai`, so eventually you can write this as `return ptime.unix_tai`. (But unless they get the speed up salobj will continue to perform its own computation, as slowness in astropy's handling of TAI was a critical problem for salobj performance).

Reply:
I can check that out when the new astropy becomes available. In the meantime, using salobj.tai_from_utc(ptime) requires all of the microseconds in my unit test to be modified in order to pass. That seems a bit weird. Can you comment on why that would be the case?

Reply:
I don't know. But microseconds is a small number when time is represented as a float, so it's not entirely surprising. I'd be tempted to test for approximate equality.

BTW: `unix_tai` is actually new AstroPy 6.1, not 6.2. https://docs.astropy.org/en/stable/changelog.html

Reply:
Done.